shared/AppInsightsCommon/src/StorageHelperFuncs.ts

```diff
@@ -30,56 +32,214 @@ function _getLocalStorageObject(): Storage {
+                                    _verifiedLocalStorage = null;
+                                } else {
+                                    _verifiedSessionStorage = null;
+                                }
+                                return null;
+                            }
+                        },
+                        setItem: function(key: string, value: string): void {
+                            try {
+                                originalStorage.setItem(key, value);
+                            } catch (e) {
+                                // Reset the cached storage on error
+                                if (storageType === StorageType.LocalStorage) {
+                                    _verifiedLocalStorage = null;
+                                } else {
+                                    _verifiedSessionStorage = null;
+                                }
+                            }
+                        },
+                        removeItem: function(key: string): void {
+                            try {
+                                originalStorage.removeItem(key);
+                            } catch (e) {
+                                // Reset the cached storage on error
+                                if (storageType === StorageType.LocalStorage) {
+                                    _verifiedLocalStorage = null;
+                                } else {
+                                    _verifiedSessionStorage = null;
+                                }
+                            }
+                        },
+                        clear: function(): void {
+                            try {
+                                originalStorage.clear();
+                            } catch (e) {
+                                // Reset the cached storage on error
+                                if (storageType === StorageType.LocalStorage) {
+                                    _verifiedLocalStorage = null;
+                                } else {
+                                    _verifiedSessionStorage = null;
+                                }
+                            }
+                        },
+                        key: function(index: number): string {
+                            try {
+                                return originalStorage.key(index);
+                            } catch (e) {
+                                // Reset the cached storage on error
+                                if (storageType === StorageType.LocalStorage) {
+                                    _verifiedLocalStorage = null;
+                                } else {
+                                    _verifiedSessionStorage = null;
+                                }
+                                return null;
+                            }
+                        },
+                        get length(): number {
```

Comment:
Only protect write operations and simplify the wrapping function to not duplicate the code